nearswap/src/twap.rs

```diff
@@ -0,0 +1,189 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::json_types::{U128, U64};
+use near_sdk::collections::{LookupMap, Vector};
+use near_sdk::serde::{Deserialize, Serialize};
+use near_sdk::{AccountId, Balance, Timestamp};
+use std::convert::{TryFrom,TryInto};
+
+use std::fmt;
+use crate::constants::*;
+
+#[derive(Clone, BorshSerialize, BorshDeserialize, Serialize, Deserialize, Copy)]
+pub struct Twap {
+    // timestamp
+    pub block_timestamp: U64,
+    // Number of observations till block_timestamp
+    pub num_of_observations: U128,
+    // cumulative price of token0 till block_timestamp
+    pub price0cumulative: U128,
+    // cumulative price of token1 till block_timestamp
+    pub price1cumulative: U128,
+}
+
+impl Twap {
+    /// @notice Transforms a previous observation into a new observation.
+    /// @dev block_timestamp _must_ be chronologically equal to or greater than last.block_timestamp, safe for 0 or 1 overflows
+    /// @param last The specified observation to be transformed
+    /// @param block_timestamp The timestamp of the new observation
+    /// @param price0 price of first token
+    /// @param price1 price of second token
+    /// @return Observation The newly populated observation
```

Comment:
This is Ethereum style Documentation. Look at other docs strings here. We don't use `@notice` nor `@param`. 
Here you have a good example: https://github.com/robert-zaremba/vostok-dao/blob/3871f679e8676472e42c7bab03c8dacdee7606a1/src/lib.rs#L40

Reply:
updated